### PR TITLE
Detect index cardinality

### PR DESCRIPTION
**Pull request by rawalpuneet**
*Issue #, if available:*

*Description of changes:*
adding branch for upcoming blog on cardinality test. This project can be used with or without the blog

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mihaialdoiu** [comment]
Please, remove the .DS_Store file from this PR.

**Comment by mihaialdoiu** [comment]
I have a few comments to this PR, that should be resolved before merging:
1. Clean-up up commented out lines, looks like over code. Or add appropriate comments if those still to be there.
https://github.com/aws-samples/amazon-documentdb-samples/pull/45/files#diff-1e190c89ce4b47926bd41b2a8a7f1cf2067975f632ea024085f0452c5bd53780R52

2. Comments for each function is a general best practice
3. It would be nice to output a progress, a message “Checking collection….“, as it sits now you just wait for the script to finish, not knowing if it’s progressing or not.
4. I see get_param() has an error:
```
python3 detect-cardinality.py --help
Traceback (most recent call last):
  File "detect-cardinality.py", line 154, in main
    get_param()
.
.
.
ValueError: incomplete format
incomplete format
```

5.  Readme says there’s a requirement for boto3, but is not, it is for pandas. Additionally, there are some spelling mistakes, `DocumnetDB` instead of `DocumentDB`.

Optional changes (if you have time):
1. Instead of global variables, should use function and return the connection object to be used by other functions
2. Catch for errors for all functions

**Comment by mihaialdoiu** [comment]
> .DS_Store

Not removed yet, make sure you configure your .gitignore, commit and push.

File: blogs/index-cardinality-detection/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+| -sample, --sample_count | Max documents to sample for each index. Increasing this limit may result in higher IOPS cost and extended execution time | 100000 | |
+
+### How to run the script 
+1. Download CA cert file
+    ```
+    wget https://truststore.pki.rds.amazonaws.com/global/global-bundle.pem
+    ```
+2. Install python dependencies 
+    ```
+    sudo pip3 install pymongo boto3
```
Comment (mihaialdoiu) [inline blogs/index-cardinality-detection/README.md:31]: pandas, not boto3

File: blogs/index-cardinality-detection/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+Amazon DocumentDB uses a B-tree data structure for its indexes. A B-tree index uses a hierarchical structure that stores data in its nodes in a sorted order. B-Trees are excellent data structure for fast retrievals when index cardinality is high. As a rule of thumb Amazon DocumentDB indexes should have less than 1% selectivity. This script accepts documentdb cluster endpoint as a parameter and generates a report across all the databases and collections for indexes lower than threshold ( Default 1% ) provided. 
+
+### Supported Parameters 
+
+### Requirements 
+* Python 3.9+ installed 
+* Pandas 
+* Mongo Client 4.0+ 
+* AWS Cli and Amazon DocumentDB
```
Comment (mihaialdoiu) [inline blogs/index-cardinality-detection/README.md:11]: Is AWS cli actually needed?

File: blogs/index-cardinality-detection/README.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
+# Detecting low cardinality indexes for DocumentDB Performance 
+
+Amazon DocumentDB uses a B-tree data structure for its indexes. A B-tree index uses a hierarchical structure that stores data in its nodes in a sorted order. B-Trees are excellent data structure for fast retrievals when index cardinality is high. As a rule of thumb Amazon DocumentDB indexes should have less than 1% selectivity. This script accepts documentdb cluster endpoint as a parameter and generates a report across all the databases and collections for indexes lower than threshold ( Default 1% ) provided. 
+
+### Supported Parameters 
+
+### Requirements 
+* Python 3.9+ installed
```
Comment (mihaialdoiu) [inline blogs/index-cardinality-detection/README.md:8]: 3.9+ or 3.x ?

File: blogs/index-cardinality-detection/README.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    --password=[PASSWORD] \
+    --sslCAFile global-bundle.pem
+
+    ```
+    Replace [DOCDB-CLUSTER-ENDPOINT], [USERNAME] and [PASSWORD] to match your DocumentDB database. 
+5. Clone this repository locally 
+    ```
+    git clone https://github.com/aws-samples/amazon-documentdb-samples.git
+    cd amazon-documentdb-samples/blogs/index-cardinality-detection
+    pip install -r requirements.txt
```
Comment (mihaialdoiu) [inline blogs/index-cardinality-detection/README.md:80]: Dependencies are installed earlier at step 2.

File: blogs/index-cardinality-detection/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,2 @@
+boto3==1.22.8
```
Comment (mihaialdoiu) [inline blogs/index-cardinality-detection/requirements.txt:1]: pandas, instead of boto3

File: blogs/index-cardinality-detection/README.md (hunk 3 above)
Comment (rawalpuneet) [reply blogs/index-cardinality-detection/README.md:8]: I tested with 3.9 but dont see anything that wont work with 3.x. I will change it. see if you get a chance to test with lower versions.

File: blogs/index-cardinality-detection/README.md (hunk 2 above)
Comment (rawalpuneet) [reply blogs/index-cardinality-detection/README.md:11]: good point. removing

File: blogs/index-cardinality-detection/README.md (hunk 1 above)
Comment (rawalpuneet) [reply blogs/index-cardinality-detection/README.md:31]: removed boto3

File: blogs/index-cardinality-detection/requirements.txt (hunk 5 above)
Comment (rawalpuneet) [reply blogs/index-cardinality-detection/requirements.txt:1]: removed boto3

**Comment by mihaialdoiu** [comment]
Looks good to me, approve. Thank you for the submission.

**Pull request merged**